code/game/objects/items/power_cells.dm

```diff
@@ -13,17 +13,250 @@
+				if(!do_after(user, delay, NONE, src, BUSY_ICON_UNSKILLED))
+					return
+
+			user.visible_message(span_notice("[user] begins manipulating [src] with [I]."),
+			span_notice("You begin rigging [src] to detonate with [I]."))
+			if(!do_after(user, delay, NONE, src, BUSY_ICON_BUILD))
+				return
+			rigged = TRUE
+			overlays += spark_overlay
+			user.visible_message(span_notice("[user] finishes manipulating [src] with [I]."),
+			span_notice("You rig [src] to explode on use with [I]."))
+		else
+			if(skill < SKILL_ENGINEER_ENGI)
+				user.visible_message(span_notice("[user] fumbles around figuring out how to manipulate [src]."),
+				span_notice("You fumble around, trying to figure out how to stabilize [src]."))
+				var/fumbling_time = SKILL_TASK_EASY
+				if(!do_after(user, fumbling_time, NONE, src, BUSY_ICON_UNSKILLED))
+					return
+				if(prob((SKILL_ENGINEER_PLASTEEL - skill) * 20))
+					to_chat(user, "<font color='danger'>After several seconds of your clumsy meddling [src] buzzes angrily as if offended. You have a <b>very</b> bad feeling about this.</font>")
+					rigged = TRUE
+					explode() //Oops. Now you fucked up (or succeeded only too well). Immediate detonation.
+			user.visible_message(span_notice("[user] begins manipulating [src] with [I]."),
+			span_notice("You begin stabilizing [src] with [I] so it won't detonate on use."))
+			if(skill > SKILL_ENGINEER_ENGI)
+				delay = max(delay - 10, 0)
+			if(!do_after(user, delay, NONE, src, BUSY_ICON_BUILD))
+				return
+			rigged = FALSE
+			overlays -= spark_overlay
+			user.visible_message(span_notice("[user] finishes manipulating [src] with [I]."),
+			span_notice("You stabilize the [src] with [I]; it will no longer detonate on use."))
+
+/obj/item/cell/proc/explode()
+	var/turf/T = get_turf(src.loc)
+/*
+* 1000-cell	explosion(T, 0, 0, 1, 1)
+* 2500-cell	explosion(T, 0, 0, 1, 1)
+* 10000-cell	explosion(T, 0, 1, 3, 3)
+* 15000-cell	explosion(T, 0, 2, 4, 4)
+* */
+	var/devastation_range = 0 //round(charge/11000)
+	var/heavy_impact_range = clamp(round(sqrt(charge) * 0.01), 0, 3)
+	var/light_impact_range = clamp(round(sqrt(charge) * 0.15), 0, 4)
+	var/flash_range = clamp(round(sqrt(charge) * 0.15), -1, 4)
+
+	explosion(T, devastation_range, heavy_impact_range, light_impact_range, 0, flash_range)
+
+	QDEL_IN(src, 1)
+
+/obj/item/cell/proc/corrupt()
+	charge /= 2
+	maxcharge /= 2
+	if (prob(10))
+		rigged = 1 //broken batterys are dangerous
+
+/obj/item/cell/emp_act(severity)
+	. = ..()
+	charge = max(charge - ((maxcharge * 0.5) / severity), 0)
+
+/obj/item/cell/ex_act(severity)
+
+	switch(severity)
+		if(EXPLODE_DEVASTATE)
+			qdel(src)
+		if(EXPLODE_HEAVY)
+			if (prob(50))
+				qdel(src)
+				return
+			if (prob(50))
+				corrupt()
+		if(EXPLODE_LIGHT)
+			if (prob(25))
+				qdel(src)
+				return
+			if (prob(25))
+				corrupt()
+		if(EXPLODE_WEAK)
+			if (prob(25))
+				corrupt()
+
+/obj/item/cell/proc/get_electrocute_damage()
```

Comment:
new procs under old procs. also you touched it so you get to autodoc it.

Reply:
Fuck. Ok this will take me a bit